### PR TITLE
Support <MatchData> response from Flags class introduced in #4894

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* https://github.com/rpherbig/dr-scripts/pull/4894 updated Flags class to return `MatchData` regex result rather than array of the string matches to support named groups
* Some scripts treated the response as an array and called methods on the result rather than accessing groups by index, and those scripts are now [breaking](https://discord.com/channels/745675889622384681/745675890242879671/831956248261296158)
* Supercedes #4895

```
--- Lich: error: undefined method `last' for #<MatchData:0x124107c0>
    combat-trainer:937:in `tend_lodged'
    combat-trainer:898:in `execute'
--- Lich: combat-trainer has exited.
```

```
--- Lich: error: undefined method first' for #<MatchData:0x10e94ea8>
    common-arcana:439:in perc_aura'
    combat-trainer:1575:in `check_starlight'
```

### Changes
* Update scripts that treated Flags result as an array to call `.to_a` on the result first
* Use named capture groups where appropriate
* Replace Flags['name'].first/last with Flags['name'][0]

### CODE REVIEW

**Comment by rcuhljr** [comment]
Looks good, a number of those don't actually need to change, but won't hurt anything. (anything with a range [i..j] returns an array on match data like expected. Ideally we'd go in and stop using .includes? on match arrays because that's basically never correct, but as a quick fix it should do.

**Comment by wstampley** [comment]
I didn't find any other cases, and these fixes look like the safe approach to me (I don't know enough about some of the names you gave capture groups, but you're usually pretty good about that stuff).

File: bescort.lic [hunk 1]
```diff
@@ -1384,7 +1384,7 @@ class Bescort
   def power_walk(pattern)
     fput 'exit' if dead? || Flags['ap-danger']
     if Flags['harness-check']
-      harness_mana([20]) unless Flags['harness-check'].include?('effortlessly')
+      harness_mana([20]) unless Flags['harness-check'].to_a.include?('effortlessly')
```
Comment (KatoakDR) [inline bescort.lic:1387]:
```suggestion
      harness_mana([20]) unless Flags['harness-check'][1] == 'effortlessly'
```

File: combat-trainer.lic [hunk 2]
```diff
@@ -211,7 +211,7 @@ class SetupProcess
   end
 
   def last_stance
-    Flags['last-stance'][0] =~ /(\d+)%.* (\d+)%.* (\d+)%.* (\d+)/
+    Flags['last-stance'].to_a[0] =~ /(\d+)%.* (\d+)%.* (\d+)%.* (\d+)/
```
Comment (KatoakDR) [inline combat-trainer.lic:214]:
```suggestion
    Flags['last-stance'][0] =~ /(\d+)%.* (\d+)%.* (\d+)%.* (\d+)/
```

File: common-arcana.lic [hunk 3]
```diff
@@ -436,7 +436,7 @@ module DRCA
     Flags.add('aura-growing?', 'Local conditions permit optimal growth of your aura', 'Local conditions are hindering the growth of your aura')
     aura = {}
     DRC.bput('perceive aura', 'Roundtime')
-    aura['level'] = Flags['aura-level'] ? starlight_messages.index(Flags['aura-level'].first) : 0
+    aura['level'] = Flags['aura-level'] ? starlight_messages.index(Flags['aura-level'].to_a.first) : 0
```
Comment (KatoakDR) [inline common-arcana.lic:439]:
```suggestion
    aura['level'] = Flags['aura-level'] ? starlight_messages.index(Flags['aura-level'][0]) : 0
```

File: forge.lic [hunk 4]
```diff
@@ -405,10 +405,10 @@ class Forge
   def assemble_part
     while Flags['forge-assembly']
       part = nil
-      part = if Flags['forge-assembly'].first == 'ready to be woven into and around the material'
+      part = if Flags['forge-assembly'].to_a.first == 'ready to be woven into and around the material'
```
Comment (KatoakDR) [inline forge.lic:408]:
```suggestion
      part = if Flags['forge-assembly'][0] == 'ready to be woven into and around the material'
```

File: mine.lic [hunk 5]
```diff
@@ -30,7 +30,7 @@ class Mine
   end
 
   def resource_level
-    case Flags['resource-level'].first
+    case Flags['resource-level'].to_a.first
```
Comment (KatoakDR) [inline mine.lic:33]:
```suggestion
    case Flags['resource-level'][0]
```

File: mm.lic [hunk 6]
```diff
@@ -74,7 +74,7 @@ class MoonMagery
     cast_spell(data, @settings)
     pause 0.25
     if Flags['locate-fail']
-      case Flags['locate-fail'].first
+      case Flags['locate-fail'].to_a.first
```
Comment (KatoakDR) [inline mm.lic:77]:
```suggestion
      case Flags['locate-fail'][0]
```

**Comment by rpherbig** [comment]
@KatoakDR looks good to me, is this ready to be merged?

**Comment by KatoakDR** [comment]
> @KatoakDR looks good to me, is this ready to be merged?

Yes, thank you. Big 👏 to @Hiinky @rcuhljr @wstampley for helping triage while I was away

**Pull request merged**